packages/inter-protocol/src/clientSupport.js

```diff
@@ -0,0 +1,224 @@
+ * @param {string} previousOffer
+ * @returns {import('@agoric/smart-wallet/src/offers.js').OfferSpec}
+ */
+export const makeAdjustOffer = (brands, opts, previousOffer) => {
+  // NB: not really a Proposal because the brands are not remotes
+  // Instead they're copyRecord like  "{"boardId":"board0257","iface":"Alleged: IST brand"}" to pass through the boardId
+  // mustMatch(harden(proposal), ProposalShape);
+  const proposal = makeVaultProposal(brands, opts);
+
+  return {
+    id: opts.offerId,
+    invitationSpec: {
+      source: 'continuing',
+      previousOffer,
+      invitationMakerName: 'AdjustBalances',
+    },
+    proposal,
+  };
+};
+
+/**
+ * @param {Record<string, import('@agoric/vats/tools/board-utils').BoardRemote>} brands
+ * @param {{ offerId: string, collateralBrandKey?: string, giveMinted: number }} opts
+ * @param {string} previousOffer
+ * @returns {import('@agoric/smart-wallet/src/offers.js').OfferSpec}
+ */
+export const makeCloseOffer = (brands, opts, previousOffer) => {
+  const proposal = makeVaultProposal(brands, opts);
+  console.warn('vaults close give', proposal.give);
+
+  return {
+    id: opts.offerId,
+    invitationSpec: {
+      source: 'continuing',
+      previousOffer,
+      invitationMakerName: 'CloseVault',
+    },
+    proposal,
+  };
+};
+
+export const Offers = {
+  vaults: {
+    OpenVault: makeOpenOffer,
+    AdjustBalances: makeAdjustOffer,
+    CloseVault: makeCloseOffer,
+  },
+};
+
+/**
+ * @param {string} vaultId
+ * @param {Promise<import('@agoric/smart-wallet/src/smartWallet').CurrentWalletRecord>} currentP
+ * @returns {Promise<string>} offer id in which the vault was made
+ */
+export const lookupOfferIdForVault = async (vaultId, currentP) => {
+  const { offerToPublicSubscriberPaths } = await currentP;
+
+  for (const [offerId, publicSubscribers] of Object.entries(
+    offerToPublicSubscriberPaths,
+  )) {
+    if (publicSubscribers.vault?.endsWith(vaultId)) {
+      return offerId;
+    }
+  }
+  throw Fail`vault ${vaultId} not found`;
+};
+
+/**
+ * @param {Record<string, import('@agoric/vats/tools/board-utils').BoardRemote>} brands
+ * @param {({ wantMinted: number | undefined, giveMinted: number | undefined })} opts
+ * @param {number} [fee=0]
+ * @param {string} [anchor]
+ * @returns {Proposal} XXX not a real proposal, uses BoardRemote
+ */
+const makePsmProposal = (brands, opts, fee = 0, anchor = 'AUSD') => {
+  const giving = opts.giveMinted ? 'minted' : 'anchor';
+  const brand =
+    giving === 'anchor'
+      ? { in: brands[anchor], out: brands.IST }
+      : { in: brands.IST, out: brands[anchor] };
+  const value =
+    Number(giving === 'anchor' ? opts.wantMinted : opts.giveMinted) *
+    Number(COSMOS_UNIT);
+  const adjusted = {
+    in: BigInt(Math.ceil(giving === 'anchor' ? value / (1 - fee) : value)),
+    out: BigInt(Math.ceil(giving === 'minted' ? value * (1 - fee) : value)),
+  };
```

Comment:
When I have multiple consecutive clauses with the same predicate, I usually find it clearer to switch to an if-else so I only check once, and all the like clauses are together. It's especially weird that the final one has the opposite polarity. 

consider a multi-return IIFE.

... OIC. This code is mostly just being moved. Feel free to ignore.